File: examples/fsm_platformer/fsm_platformer_python/jump_command.py
"""Jump Command — applies upward impulse when grounded."""

from __future__ import annotations

from src.engine.math.vector import Vector2

from fsm_platformer_python.command import Command

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from fsm_platformer_python.player_input_handler import PlayerInputHandler


class JumpCommand(Command):
    """Applies vertical jump force. Only valid when grounded."""

    def __init__(self, player_input_handler: PlayerInputHandler) -> None:
        super().__init__(player_input_handler)

    def is_valid(self) -> bool:
        return self.player_input_handler.is_grounded

    def do_before_entering(self) -> None:
        pass

    def act(self) -> None:
        player = self.player_input_handler
        if player.is_grounded:
            rb = player.rb
            rb.velocity = Vector2(rb.velocity.x, player.jump_force)

File: examples/fsm_platformer/fsm_platformer_unity/JumpCommand.cs
using UnityEngine;

// Unity APIs used: MonoBehaviour, Rigidbody2D, Vector2
public class JumpCommand : Command
{
    public JumpCommand(PlayerInputHandler handler) : base(handler) { }

    public override bool IsValid()
    {
        return playerInputHandler.IsGrounded;
    }

    public override void DoBeforeEntering() { }

    public override void Act()
    {
        if (playerInputHandler.IsGrounded)
        {
            playerInputHandler.Rb.linearVelocity = new Vector2(
                playerInputHandler.Rb.linearVelocity.x,
                playerInputHandler.JumpForce);
        }
    }
}
